pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -822,6 +845,53 @@ func (pl *DynamicResources) PreBind(ctx context.Context, cs fwk.CycleState, pod
 			state.claims[index] = claim
 		}
 	}
+
+	if !pl.enableDeviceBindingConditions || !pl.enableDeviceStatus {
+		// If we get here, we know that reserving the claim for
+		// the pod worked and we can proceed with binding it.
+		return nil
+	}
+
+	// We need to check if the device is attached to the node.
+	needWait := hasBindingConditions(state)
+
+	// If no device needs to be prepared, we can return early.
+	if !needWait {
+		return nil
+	}
+
+	// We need to decide how long we should wait for the device to be attached to the node.
```

Comment:
Nit: let's use terms from the API instead of assuming what the driver does under the hood.

```suggestion
	// We need to decide how long we should wait for the driver to set the binding conditions.
```